### PR TITLE
Add section heads to help commands.

### PR DESCRIPTION
**Pull request by richsalz**
Fixes: #9936 

This is a WIP.  When done I'll remove the TODO file.

### CODE REVIEW

**Comment by richsalz** [comment]
If someone can take a look at the output, I'd like an opinion of there should be blank line before each section.

**Comment by richsalz** [comment]
First pass done; all programs with >=20 help lines: asn1parse ca cms crl dgst ecparam enc ocsp pkcs12 pkcs8 pkeyutl req rsa rsautl s_client s_server smime speed ts verify x509

**Comment by richsalz** [comment]
All commands done, out of WIP, ready for review.

**Comment by richsalz** [comment]
in order to entice folks to look at and review this, here's some output:
```
; ./apps/openssl rsa --help
Usage: rsa [options]

General options:
 -help              Display this summary
 -check             Verify key consistency
 -*                 Any supported cipher
 -engine val        Use engine, possibly a hardware device

Input options:
 -in val            Input file
 -inform format     Input format, one of DER PEM
 -pubin             Expect a public key in input file
 -RSAPublicKey_in   Input is an RSAPublicKey
 -passin val        Input file pass phrase source

Output options:
 -out outfile       Output file
 -outform format    Output format, one of DER PEM PVK
 -pubout            Output a public key
 -RSAPublicKey_out  Output is an RSAPublicKey
 -passout val       Output file pass phrase source
 -noout             Don't print key out
 -text              Print the key in text
 -modulus           Print the RSA key modulus

PVK options:
 -pvk-strong        Enable 'Strong' PVK encoding level (default)
 -pvk-weak          Enable 'Weak' PVK encoding level
 -pvk-none          Don't enforce PVK encoding
; 
; ./apps/openssl rand --help
Usage: rand [flags] num

General options:
 -help               Display this summary
 -engine val         Use engine, possibly a hardware device

Output options:
 -out outfile        Output file
 -base64             Base64 encode output
 -hex                Hex encode output

Random state options:
 -rand val           Load the file(s) into the random number generator
 -writerand outfile  Write random data to the specified file
```

For common options (OPT_[VSXR]_OPTIONS in ```apps/lib/opt.c```), as shown in the "Random state options" text above, the prolog is common text.

File: apps/fipsinstall.c [hunk 1]
```diff
@@ -32,18 +32,26 @@ typedef enum OPTION_choice {
 } OPTION_CHOICE;
 
 const OPTIONS fipsinstall_options[] = {
+    OPT_SECTION("General"),
     {"help", OPT_HELP, '-', "Display this summary"},
+#if 0
+    /* Gross ugly long line that belongs in the manpage. */
```
Comment (t8m) [inline apps/fipsinstall.c]: Just delete it and add it to the manpage unless the example is already there.

File: apps/pkcs8.c [hunk 2]
```diff
@@ -31,30 +31,37 @@ typedef enum OPTION_choice {
 
 const OPTIONS pkcs8_options[] = {
     {"help", OPT_HELP, '-', "Display this summary"},
```
Comment (t8m) [inline apps/pkcs8.c]: Missing General options section marker.

File: apps/fipsinstall.c (hunk 1 above)
Comment (richsalz) [reply apps/fipsinstall.c]: the manpage has a couple of complete examples so i removed this.

File: apps/pkcs8.c (hunk 2 above)
Comment (richsalz) [reply apps/pkcs8.c]: good catch!

**Comment by richsalz** [comment]
fixup commit pushed.

File: apps/ciphers.c [hunk 3]
```diff
@@ -31,31 +31,34 @@ typedef enum OPTION_choice {
...
+    OPT_SECTION("General"),
     {"help", OPT_HELP, '-', "Display this summary"},
+
+    OPT_SECTION("Output"),
     {"v", OPT_V, '-', "Verbose listing of the SSL/TLS ciphers"},
     {"V", OPT_UPPER_V, '-', "Even more verbose"},
     {"s", OPT_S, '-', "Only supported ciphers"},
 #ifndef OPENSSL_NO_SSL3
-    {"ssl3", OPT_SSL3, '-', "SSL3 mode"},
+    {"ssl3", OPT_SSL3, '-', "Ciphers defined for SSL3"},
```
Comment (levitte) [inline apps/ciphers.c]: Is this, and the similar ones following, really output options per se?  They rather seem like selectors

File: apps/asn1pars.c [hunk 4]
```diff
@@ -27,27 +27,32 @@ typedef enum OPTION_choice {
 } OPTION_CHOICE;
 
 const OPTIONS asn1parse_options[] = {
+    OPT_SECTION("General"),
     {"help", OPT_HELP, '-', "Display this summary"},
+    {"oid", OPT_OID, '<', "file of extra oid definitions"},
+
+    OPT_SECTION("I/O"),
```
Comment (levitte) [inline apps/asn1pars.c]: In others, you have separated Input from Output.  Not here, is there a reason?

File: apps/ca.c [hunk 5]
```diff
@@ -154,10 +154,30 @@ typedef enum OPTION_choice {
...
 
 const OPTIONS ca_options[] = {
+    OPT_SECTION("General"),
     {"help", OPT_HELP, '-', "Display this summary"},
     {"verbose", OPT_VERBOSE, '-', "Verbose output during processing"},
+    {"outdir", OPT_OUTDIR, '/', "Where to put output cert"},
+    {"in", OPT_IN, '<', "The input PEM encoded cert request(s)"},
+    {"infiles", OPT_INFILES, '-', "The last argument, requests to process"},
+    {"out", OPT_OUT, '>', "Where to put the output file(s)"},
+    {"notext", OPT_NOTEXT, '-', "Do not print the generated certificate"},
```
Comment (levitte) [inline apps/ca.c]: For other commands, the input and output options have their own sections, but not this one:

File: apps/asn1pars.c [hunk 6]
```diff
@@ -27,27 +27,32 @@ typedef enum OPTION_choice {
...
     {"genconf", OPT_GENCONF, 's', "file to generate ASN1 structure from"},
-    {OPT_MORE_STR, 0, 0, "(-inform  will be ignored)"},
     {"strictpem", OPT_STRICTPEM, 0,
      "do not attempt base64 decode outside PEM markers"},
     {"item", OPT_ITEM, 's', "item to parse and print"},
+    {OPT_MORE_STR, 0, 0, "(-inform  will be ignored)"},
+
+    OPT_SECTION("Formatting"),
+    {"i", OPT_INDENT, 0, "indents the output"},
+    {"noout", OPT_NOOUT, 0, "do not produce any output"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/asn1pars.c]: fixed.

File: apps/ciphers.c (hunk 3 above)
Comment (richsalz) [reply apps/ciphers.c]: Good point; I changed it to "Cipher selection"

**Comment by richsalz** [comment]
As for the Input, Output, Input/Output: if there were a bunch of options, or I thought it made sense, I divided them into sections, otherwise I didn't.  I agree it might be better to go for consistency, but I also dislike a section that has only one option (yes sometimes that is unavoidable).

I hope that this topic can be addressed in a separate PR.  (Probably by someone else :)

File: apps/ciphers.c [hunk 7]
```diff
@@ -34,7 +34,7 @@ const OPTIONS ciphers_options[] = {
     OPT_SECTION("General"),
     {"help", OPT_HELP, '-', "Display this summary"},
 
-    OPT_SECTION("Output"),
+    OPT_SECTION("Cipher specification"),
```
Comment (t8m) [inline apps/ciphers.c]: Hmm, should there be two separate groups? Output (-v, -V, -stdname, -convert) and Cipher selection (for the rest)?

Comment (richsalz) [reply apps/ciphers.c]: yes, there should :)  additional commit pushed.  thanks.

**Comment by t8m** [comment]
My approval still holds.

File: apps/ca.c (hunk 5 above)
Comment (richsalz) [reply apps/ca.c]: see https://github.com/openssl/openssl/pull/9953#issuecomment-539992422

File: apps/asn1pars.c (hunk 4 above)
Comment (richsalz) [reply apps/asn1pars.c]: see https://github.com/openssl/openssl/pull/9953#issuecomment-539992422

**Comment by t8m** [comment]
ping @openssl/omc for 2nd review

**Comment by richsalz** [comment]
Another sample, to entice reviewers:
```
; ./apps/openssl pkcs7 --help
Usage: pkcs7 [options]

General options:
 -help             Display this summary
 -engine val       Use engine, possibly a hardware device

Input options:
 -in infile        Input file
 -inform PEM|DER   Input format - DER or PEM

Output options:
 -outform PEM|DER  Output format - DER or PEM
 -out outfile      Output file
 -noout            Don't output encoded data
 -text             Print full details of certificates
 -print            Print out all fields of the PKCS7 structure
 -print_certs      Print_certs  print any certs or crl in the input
; 
```

**Comment by levitte** [comment]
Travis failures seem relevant, there seems to be some option duplication in some commands.

**Comment by t8m** [comment]
Yeah, the rebase broke the apps/enc.c. There are added duplicate options in, out, and pass.

**Comment by t8m** [comment]
OK, re-approved now.
ping @levitte ?

**Comment by richsalz** [comment]
Another sample:
```
; ./apps/openssl ocsp --help
Usage: ocsp [options]

General options:
 -help                   Display this summary
 -ignore_err             Ignore error on OCSP request or response and continue running
 -CAfile infile          Trusted certificates file
 -CApath infile          Trusted certificates directory
 -no-CAfile              Do not load the default certificates file
 -no-CApath              Do not load certificates from the default certificates directory

Responder options:
 -timeout +int           Connection timeout (in seconds) to the OCSP responder
 -resp_no_certs          Don't include any certificates in response
 -multi +int             run multiple responder processes
 -no_certs               Don't include any certificates in signed request
 -badsig                 Corrupt last byte of loaded OSCP response signature (for test)
 -CA infile              CA certificate
 -nmin +int              Number of minutes before next update
 -nrequest +int          Number of requests to accept (default unlimited)
 -reqin val              File with the DER-encoded request
 -signer infile          Certificate to sign OCSP request with
 -sign_other infile      Additional certificates to include in signed request
 -index infile           Certificate status index file
 -ndays +int             Number of days before next update
 -rsigner infile         Responder certificate to sign responses with
 -rkey infile            Responder key to sign responses with
 -rother infile          Other certificates to include in response
 -rmd val                Digest Algorithm to use in signature of OCSP response
 -rsigopt val            OCSP response signature parameter in n:v form
 -header val             key=value header to add
 -rcid val               Use specified algorithm for cert id in response
 -*                      Any supported digest algorithm (sha1,sha256, ... )

Client options:
 -url val                Responder URL
 -host val               TCP/IP hostname:port to connect to
 -port +int              Port to run responder on
 -out outfile            Output filename
 -noverify               Don't verify response at all
 -nonce                  Add OCSP nonce to request
 -no_nonce               Don't add OCSP nonce to request
 -no_signature_verify    Don't check signature on response
 -resp_key_id            Identify response by signing certificate key ID
 -no_cert_verify         Don't check signing certificate
 -text                   Print text form of request and response
 -req_text               Print text form of request
 -resp_text              Print text form of response
 -no_chain               Don't chain verify response
 -no_cert_checks         Don't do additional checks on signing certificate
 -no_explicit            Do not explicitly check the chain, just verify the root
 -trust_other            Don't verify additional certificates
 -no_intern              Don't search certificates contained in response for signer
 -respin val             File with the DER-encoded response
 -VAfile infile          Validator certificates file
 -verify_other infile    Additional certificates to search for signer
 -path val               Path to use in OCSP request
 -cert infile            Certificate to check
 -serial val             Serial number to check
 -validity_period ulong  Maximum validity discrepancy in seconds
 -signkey val            Private key to sign OCSP request with
 -reqout val             Output file for the DER-encoded request
 -respout val            Output file for the DER-encoded response
 -issuer infile          Issuer certificate
 -status_age +int        Maximum status age in seconds

Validation options:
 -policy val             adds policy to the acceptable policy set
 -purpose val            certificate chain purpose
 -verify_name val        verification policy name
 -verify_depth int       chain depth limit
 -auth_level int         chain authentication security level
 -attime intmax          verification epoch time
 -verify_hostname val    expected peer hostname
 -verify_email val       expected peer email
 -verify_ip val          expected peer IP address
 -ignore_critical        permit unhandled critical extensions
 -issuer_checks          (deprecated)
 -crl_check              check leaf certificate revocation
 -crl_check_all          check full chain revocation
 -policy_check           perform rfc5280 policy checks
 -explicit_policy        set policy variable require-explicit-policy
 -inhibit_any            set policy variable inhibit-any-policy
 -inhibit_map            set policy variable inhibit-policy-mapping
 -x509_strict            disable certificate compatibility work-arounds
 -extended_crl           enable extended CRL features
 -use_deltas             use delta CRLs
 -policy_print           print policy processing diagnostics
 -check_ss_sig           check root CA self-signatures
 -trusted_first          search trust store first (default)
 -suiteB_128_only        Suite B 128-bit-only mode
 -suiteB_128             Suite B 128-bit mode allowing 192-bit algorithms
 -suiteB_192             Suite B 192-bit-only mode
 -partial_chain          accept chains anchored by intermediate trust-store CAs
 -no_alt_chains          (deprecated)
 -no_check_time          ignore certificate validity time
 -allow_proxy_certs      allow the use of proxy certificates
; 
```

**Comment by t8m** [comment]
For the record my approval still holds.

**Comment by richsalz** [comment]
I had to rebase this because #8442 added new flags to several commands.  There were, of course, merge conflicts.  No other changes were made.

Seeking reviews.  First opened September 19.  No feedback (except poor patient @t8m :) for nearly a month.

**Comment by t8m** [comment]
My approval still holds. It would be nice if @openssl/omc provided the second review soon to avoid further merge conflicts.

File: apps/ciphers.c [hunk 8]
```diff
@@ -31,33 +31,38 @@ typedef enum OPTION_choice {
...
-    {"tls1", OPT_TLS1, '-', "TLS1 mode"},
+    {"tls1", OPT_TLS1, '-', "Ciphers defined for TLS1"},
 #endif
 #ifndef OPENSSL_NO_TLS1_1
-    {"tls1_1", OPT_TLS1_1, '-', "TLS1.1 mode"},
+    {"tls1_1", OPT_TLS1_1, '-', "Ciphers defined for TLS1.1"},
 #endif
 #ifndef OPENSSL_NO_TLS1_2
-    {"tls1_2", OPT_TLS1_2, '-', "TLS1.2 mode"},
+    {"tls1_2", OPT_TLS1_2, '-', "Ciphers defined for TLS1.2"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/ciphers.c]: done.

File: apps/ciphers.c [hunk 9]
```diff
@@ -31,33 +31,38 @@ typedef enum OPTION_choice {
...
 #endif
 #ifndef OPENSSL_NO_TLS1_3
-    {"tls1_3", OPT_TLS1_3, '-', "TLS1.3 mode"},
+    {"tls1_3", OPT_TLS1_3, '-', "Ciphers defined for TLS1.3"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/ciphers.c]: done.

File: apps/ciphers.c [hunk 10]
```diff
@@ -31,33 +31,38 @@ typedef enum OPTION_choice {
...
+
+    OPT_SECTION("Cipher specification"),
     {"s", OPT_S, '-', "Only supported ciphers"},
 #ifndef OPENSSL_NO_SSL3
-    {"ssl3", OPT_SSL3, '-', "SSL3 mode"},
+    {"ssl3", OPT_SSL3, '-', "Ciphers defined for SSL3"},
 #endif
 #ifndef OPENSSL_NO_TLS1
-    {"tls1", OPT_TLS1, '-', "TLS1 mode"},
+    {"tls1", OPT_TLS1, '-', "Ciphers defined for TLS1"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/ciphers.c]: done.

File: apps/ocsp.c [hunk 11]
```diff
@@ -147,31 +147,69 @@ typedef enum OPTION_choice {
...
-     "Connection timeout (in seconds) to the OCSP responder"},
-    {"url", OPT_URL, 's', "Responder URL"},
-    {"host", OPT_HOST, 's', "TCP/IP hostname:port to connect to"},
-    {"port", OPT_PORT, 'p', "Port to run responder on"},
     {"ignore_err", OPT_IGNORE_ERR, '-',
      "Ignore error on OCSP request or response and continue running"},
-    {"noverify", OPT_NOVERIFY, '-', "Don't verify response at all"},
-    {"nonce", OPT_NONCE, '-', "Add OCSP nonce to request"},
-    {"no_nonce", OPT_NO_NONCE, '-', "Don't add OCSP nonce to request"},
+    {"CAfile", OPT_CAFILE, '<', "Trusted certificates file"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/ocsp.c]: I did each command without thinking too much of overall consistency.  I agree that more shuffling around can and should be done.  I would rather see that done as a followup PR.

File: apps/req.c [hunk 12]
```diff
@@ -140,18 +126,41 @@ const OPTIONS req_options[] = {
...
+    {"keyform", OPT_KEYFORM, 'f', "Key file format"},
+    {"pubkey", OPT_PUBKEY, '-', "Output public key"},
+    {"keyout", OPT_KEYOUT, '>', "File to send the key to"},
+    {"passin", OPT_PASSIN, 's', "Private key password source"},
+    {"passout", OPT_PASSOUT, 's', "Output file pass phrase source"},
+    {"newkey", OPT_NEWKEY, 's', "Specify as type:bits"},
+    {"pkeyopt", OPT_PKEYOPT, 's', "Public key options as opt:value"},
+    {"sigopt", OPT_SIGOPT, 's', "Signature parameter in n:v form"},
+    {"verify", OPT_VERIFY, '-', "Verify signature on REQ"},
+    {"modulus", OPT_MODULUS, '-', "RSA modulus"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/req.c]: fixed thanks.

File: apps/req.c [hunk 13]
```diff
@@ -140,18 +126,41 @@ const OPTIONS req_options[] = {
...
+    {"key", OPT_KEY, 's', "Private key to use"},
+    {"keyform", OPT_KEYFORM, 'f', "Key file format"},
+    {"pubkey", OPT_PUBKEY, '-', "Output public key"},
+    {"keyout", OPT_KEYOUT, '>', "File to send the key to"},
+    {"passin", OPT_PASSIN, 's', "Private key password source"},
+    {"passout", OPT_PASSOUT, 's', "Output file pass phrase source"},
+    {"newkey", OPT_NEWKEY, 's', "Specify as type:bits"},
+    {"pkeyopt", OPT_PKEYOPT, 's', "Public key options as opt:value"},
+    {"sigopt", OPT_SIGOPT, 's', "Signature parameter in n:v form"},
+    {"verify", OPT_VERIFY, '-', "Verify signature on REQ"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/req.c]: I moved it to the "general" section.  Not great, admittedly, but better.

I also moved in/inform there, and out/outform  to the "output" section.

File: apps/s_client.c [hunk 14]
```diff
@@ -664,51 +691,37 @@ const OPTIONS s_client_options[] = {
...
-     "Offer SRTP key management with a colon-separated profile list"},
-#endif
-    {"keymatexport", OPT_KEYMATEXPORT, 's',
-     "Export keying material using label"},
-    {"keymatexportlen", OPT_KEYMATEXPORTLEN, 'p',
-     "Export len bytes of keying material (default 20)"},
-    {"maxfraglen", OPT_MAXFRAGLEN, 'p',
-     "Enable Maximum Fragment Length Negotiation (len values: 512, 1024, 2048 and 4096)"},
-    {"fallback_scsv", OPT_FALLBACKSCSV, '-', "Send the fallback SCSV"},
     {"name", OPT_PROTOHOST, 's',
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/s_client.c]: yes, that's better.  moved.

File: apps/ocsp.c (hunk 11 above)
[the remark this answers is not in the document]
Comment (vdukhovni) [reply apps/ocsp.c]: Is that PR forthcoming? I am somewhat reluctant to have this merged if the followup PR may not materialize.

[the remark this answers is not in the document]
Comment (richsalz) [reply apps/ocsp.c]: While I am likely to look at this, I make no commitment.  I think this PR, *as it currently stands* is a big improvement over what OpenSSL currently has.  I will shrug my shoulders (and scratch my head in wonderment :) if you don't think so, too.

File: apps/s_time.c [hunk 15]
```diff
@@ -92,6 +79,25 @@ const OPTIONS s_time_options[] = {
...
+    {"verify", OPT_VERIFY, 'p',
+     "Turn on peer certificate verification, set depth"},
+    {"time", OPT_TIME, 'p', "Seconds to collect data, default " SECONDSSTR},
+    {"www", OPT_WWW, 's', "Fetch specified page from the site"},
+
+    OPT_SECTION("Certificate"),
+    {"nameopt", OPT_NAMEOPT, 's', "Various certificate name options"},
+    {"cert", OPT_CERT, '<', "Cert file to use, PEM format assumed"},
+    {"key", OPT_KEY, '<', "File with key, PEM; default is -cert file"},
+    {"cafile", OPT_CAFILE, '<', "PEM format file of CA's"},
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/s_time.c]: Same; I am not doing this now.

File: apps/smime.c [hunk 16]
```diff
@@ -52,39 +52,62 @@ const OPTIONS smime_options[] = {
...
+    {"recip", OPT_RECIP, '<', "Recipient certificate file for decryption"},
+
+    OPT_SECTION("Email"),
     {"to", OPT_TO, 's', "To address"},
     {"from", OPT_FROM, 's', "From address"},
     {"subject", OPT_SUBJECT, 's', "Subject"},
     {"text", OPT_TEXT, '-', "Include or delete text MIME headers"},
+    {"nosmimecap", OPT_NOSMIMECAP, '-', "Omit the SMIMECapabilities attribute"},
+
+    OPT_SECTION("Certificate chain"),
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/smime.c]: Same reply, I am not doing this for this PR.

File: apps/x509.c [hunk 17]
```diff
@@ -65,37 +65,60 @@ typedef enum OPTION_choice {
...
+    {"ext", OPT_EXT, 's', "Print various X509V3 extensions"},
+    {"C", OPT_C, '-', "Print out C code forms"},
+#ifndef OPENSSL_NO_MD5
+    {"subject_hash_old", OPT_SUBJECT_HASH_OLD, '-',
+     "Print old-style (MD5) issuer hash value"},
+    {"issuer_hash_old", OPT_ISSUER_HASH_OLD, '-',
+     "Print old-style (MD5) subject hash value"},
+#endif
+
+    OPT_SECTION("Certificate"),
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/x509.c]: The only output-related option I could see is "-nameopt" which I moved to output.  Did you8 have others in mind?

File: apps/x509.c [hunk 18]
```diff
@@ -104,51 +127,35 @@ const OPTIONS x509_options[] = {
...
     {"extensions", OPT_EXTENSIONS, 's', "Section from config file to use"},
     {"nameopt", OPT_NAMEOPT, 's', "Various certificate name options"},
     {"certopt", OPT_CERTOPT, 's', "Various certificate text options"},
     {"checkhost", OPT_CHECKHOST, 's', "Check certificate matches host"},
     {"checkemail", OPT_CHECKEMAIL, 's', "Check certificate matches email"},
     {"checkip", OPT_CHECKIP, 's', "Check certificate matches ipaddr"},
+    {"force_pubkey", OPT_FORCE_PUBKEY, '<', "Force the key to put inside certificate"},
+    {"subj", OPT_SUBJ, 's', "Set or override certificate subject (and issuer)"},
+
+    OPT_SECTION("CA"),
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/x509.c]: Can you give me more details?  Which options should move, and to which section?

File: apps/verify.c [hunk 19]
```diff
@@ -36,10 +36,17 @@ typedef enum OPTION_choice {
...
+    OPT_SECTION("General"),
     {"help", OPT_HELP, '-', "Display this summary"},
+#ifndef OPENSSL_NO_ENGINE
+    {"engine", OPT_ENGINE, 's', "Use engine, possibly a hardware device"},
+#endif
     {"verbose", OPT_VERBOSE, '-',
         "Print extra information about the operations being performed."},
+    {"nameopt", OPT_NAMEOPT, 's', "Various certificate name options"},
+
+    OPT_SECTION("Certificate chain"),
```
[the remark this answers is not in the document]
Comment (richsalz) [reply apps/verify.c]: Same reply, I am not doing this for this PR.

**Comment by t8m** [comment]
I believe we should focus on real errors/mistakes and not try to achieve perfection in this PR. IMO  even in the state as it is now it would be major improvement over the current situation.

**Comment by richsalz** [comment]
I don't understand your question.

**Comment by richsalz** [comment]
(PS:  Want to tweak the labels so the 24hour hold timer works?)

**Comment by paulidale** [comment]
I'm wondering why there are OPT_MORE_STR lines in the text.  They seem to be manually putting line breaks in places -- this could be automated.

It's quite poissible I missing something egregious.

**Comment by richsalz** [comment]
The number really didn't change.  Auto-wrapping the text would be more trouble than it's worth for the roughly-a-dozen lines:
```
; g m
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
; g grep OPT_MORE_STR | wc 
     14     105     859
; g co app-help-categories 
Switched to branch 'app-help-categories'
; g grep OPT_MORE_STR | wc 
     13      91     750
; 
```

**Comment by paulidale** [comment]
Merged to master.  Thanks for the effort and endurance on of this one.

**Pull request closed without merging**